File: Vanilla/sources/zdoc/media/lua/shared/Library/BarricadeAble.lua
---@class BarricadeAble : zombie.iso.objects.interfaces.BarricadeAble
BarricadeAble = {}

---@public
---@return boolean
function BarricadeAble:isBarricadeAllowed() end

---@public
---@return boolean
function BarricadeAble:isBarricaded() end

---@public
---@return IsoBarricade
function BarricadeAble:getBarricadeOnOppositeSquare() end

---@public
---@return IsoGridSquare
function BarricadeAble:getOppositeSquare() end

---@public
---@return IsoBarricade
function BarricadeAble:getBarricadeOnSameSquare() end

---@public
---@param arg0 IsoGameCharacter
---@return IsoBarricade
function BarricadeAble:getBarricadeForCharacter(arg0) end

---@public
---@return boolean
function BarricadeAble:getNorth() end

---@public
---@param arg0 IsoGameCharacter
---@return IsoBarricade
function BarricadeAble:getBarricadeOppositeCharacter(arg0) end

---@public
---@return IsoGridSquare
function BarricadeAble:getSquare() end

File: Vanilla/sources/zomboid/zombie/iso/objects/interfaces/BarricadeAble.java
package zombie.iso.objects.interfaces;

import zombie.characters.IsoGameCharacter;
import zombie.iso.IsoGridSquare;
import zombie.iso.objects.IsoBarricade;

public interface BarricadeAble {
   boolean isBarricaded();

   boolean isBarricadeAllowed();

   IsoBarricade getBarricadeOnSameSquare();

   IsoBarricade getBarricadeOnOppositeSquare();

   IsoBarricade getBarricadeForCharacter(IsoGameCharacter var1);

   IsoBarricade getBarricadeOppositeCharacter(IsoGameCharacter var1);

   IsoGridSquare getSquare();

   IsoGridSquare getOppositeSquare();

   boolean getNorth();
}
